contrib/babelfishpg_tds/src/backend/tds/tdssecure.c

```diff
@@ -91,18 +92,24 @@ SslHandShakeRead(BIO *h, char *buf, int size)
 					/* Read the complete remaining of the header and throw away the bytes */
 					while(res < remainingRead)
 					{
-						res += secure_raw_read(((Port *) BIO_get_data(h)), tempBuf,
-								remainingRead - res);
+						int tmp_res = 0;
+						if ((tmp_res = SslRead(h, tempBuf, remainingRead - res)) <= 0)
+						{
+							return tmp_res;
+						}
+						res += tmp_res;
 					}
 
 					/*
 					 * Read the actual data and return the res of the actual data read
 					 * Don't worry if complete read, Openssl library will take care
 					 */
-					res = secure_raw_read(((Port *) BIO_get_data(h)), buf, size);
+					if ((res = SslRead(h, buf, size)) <= 0)
```

Comment:
why change from secure_raw_read to Ssloread?

Reply:
Because SslRead is safe wrapper around secure_raw_read()